reader/components/header/Header.jsx

```diff
@@ -1,41 +1,35 @@
-import React, { useContext } from 'react'
+import React from 'react'
 import { Button } from 'reactstrap'
 import ReactToPrint from 'react-to-print'
 import Icon from '../icons/Icon'
-import GlobalContext from '../../store/configureContext'
 import { downloadPDF } from '../pdf-downloader/PDFDownloader'
 import './Header.scss'
 import Logo from '../logo/Logo'
+import {
+  toggleOutlineSidebar,
+  toggleThumbnailsSidebar,
+} from '../../store/ui/actions'
+import { useGlobalStore } from '../../store'
 
-const Header = () => {
-  const {
-    state: {
-      pdfDocument: { pdfDocumentProxy, pdfPagesLoaded },
-      pdfMetadata,
-      isThumbnailViewVisible,
-      isOutlineViewVisible,
-      printContainerRef,
-    },
-    toggleIsThumbnailViewVisible,
-    toggleIsOutlineViewVisible,
-  } = useContext(GlobalContext)
+const Header = ({ printContainerRef }) => {
+  const [{ metadata, ui, document }, dispatch] = useGlobalStore()
 
   return (
     <div className="header">
       <div className="item d-flex justify-content-start">
         <Button
           title="Show Document Outline"
           color="none"
-          active={isOutlineViewVisible}
-          onClick={toggleIsOutlineViewVisible}
+          active={ui.isOutlineSidebarVisible}
```

Comment:
UI should not be (in) a state of an application. The application state defines an UI.

Reply:
That's true. How would you do it? Could you share some ideas or even sample code?

Reply:
I would define application state as fat _Model_ that contains actual data storage and it's mutations. For example `loading` might be indication of loading state, `document` might be a metadata collection (don't punch me about naming).

On the other side I would create a central UI component like an `Application` or `Layout` that actually defines the layout (only) but is connected to the Model. Let's call it _View_.

The View defines it's own state. For example, `sidebarTabOpened` may be one of `['outline', 'thumbnails']` or `null` what means the sidebar is closed. The parent component passes this state to child Sidebar component that is a pure function as well as other child components.

Does this make sense?

Reply:
Kind of... You know people learn most from the others and it's a shame you don't have much time to work on this. It would be nice to have look at code quality from a different points of view. And I mean not just review comments (which I really appreciate) but from real code contribution. Anyway, I will keep it in mind and try to implement it on my own step by step.

Reply:
Sorry, we might misunderstood each other. I do not ask you to change anything. I like your implementation

Reply:
Not at all. What I tried to say was that I really like your ideas and it would be great to see your contribution to code more but I understand you have another work to do.

I am not going to drastically rewrite anything. I just keep in mind your comments and I always try to implement them whenever I touch that part of code or have time. :)

Reply:
Just to extend, I express these thoughts because this is what I learned on my experience. I feel it was really helpful for me and might be interesting for you. However, it should not. Also, I have not learned some things that you do and I have something to learn from you too.

I am keen to work on other projects together. I would like to put some of my thoughts above into UI library.